File: minipress.core/minipress.appli/src/models/Article.php
<?php
declare(strict_types=1);

namespace minipress\appli\models;

use Illuminate\Database\Eloquent\Model;

class Article extends Model
{
    protected $table = 'article';
    public $timestamps = false;
    protected $casts = ['date_creation' => 'datetime', 'publie' => 'boolean'];
    protected $fillable = ['titre', 'resume', 'contenu', 'image', 'date_creation', 'publie', 'auteur_id', 'categorie_id'];

    public function auteur()
    {
        return $this->belongsTo(Utilisateur::class, 'auteur_id');
    }

    public function categorie()
    {
        return $this->belongsTo(Categorie::class, 'categorie_id');
    }
}

File: minipress.app/lib/models/article.dart
import 'auteur.dart';
import 'categorie.dart';

class Article {
  final int id;
  final String titre;
  final String? resume;
  final String contenu;
  final DateTime dateCreation;
  final Auteur auteur;
  final Categorie? categorie;
  final String? image;

  const Article({
    required this.id,
    required this.titre,
    this.resume,
    required this.contenu,
    required this.dateCreation,
    required this.auteur,
    this.categorie,
    this.image,
  });

  factory Article.fromJson(Map<String, dynamic> json) => Article(
        id: json['id'] as int,
        titre: json['titre'] as String,
        resume: json['resume'] as String?,
        contenu: json['contenu'] as String,
        dateCreation: DateTime.parse(json['date_creation'] as String),
        auteur: Auteur.fromJson(json['auteur'] as Map<String, dynamic>),
        categorie: json['categorie'] != null
            ? Categorie.fromJson(json['categorie'] as Map<String, dynamic>)
            : null,
        image: json['image'] as String?,
      );
}
